src/main/java/ch/njol/skript/bukkitutil/EnchantmentUtils.java

```diff
@@ -18,26 +18,106 @@
  */
 package ch.njol.skript.bukkitutil;
 
+import ch.njol.skript.localization.Language;
 import org.bukkit.NamespacedKey;
+import org.bukkit.Registry;
 import org.bukkit.enchantments.Enchantment;
 import org.eclipse.jdt.annotation.Nullable;
 
-import com.google.common.collect.BiMap;
-import com.google.common.collect.HashBiMap;
 import ch.njol.skript.Skript;
 
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
 /**
  * Maps enchantments to their keys.
  */
 public class EnchantmentUtils {
 
+	private static final Map<Enchantment, String> NAMES = new HashMap<>();
+	private static final Map<String, Enchantment> PATTERNS = new HashMap<>();
+	private static final boolean HAS_REGISTRY = Skript.classExists("org.bukkit.Registry") && Skript.fieldExists(Registry.class, "ENCHANTMENT");
+
+	static {
+		Language.addListener(() -> {
+			NAMES.clear();
+			PATTERNS.clear();
+			List<Enchantment> enchantments = new ArrayList<>();
+			if (HAS_REGISTRY) {
+				Registry.ENCHANTMENT.forEach(enchantments::add);
+			} else {
+				enchantments.addAll(Arrays.asList(Enchantment.values()));
+			}
+			for (Enchantment enchantment : enchantments) {
+				NamespacedKey key = enchantment.getKey();
+				final String[] names = Language.getList("enchantments." + key.getKey());
+
+				if (!names[0].startsWith("enchantments.")) {
+					NAMES.put(enchantment, names[0]);
```

Comment:
We could also choose to store names for undefined enchantments in this map (e.g. the full namespaced key as done in toString)